github_issue_summarization/distributed/train.py

```diff
@@ -14,164 +15,208 @@
+  decoder_input_data, decoder_target_data = load_decoder_inputs(data_dir + 'train_title_vecs.npy')
 
-while True:
-  if os.path.isfile(data_dir + 'train_body_vecs.npy'):
-    break
-  print("Waiting for dataset")
-  time.sleep(2)
-encoder_input_data, doc_length = load_encoder_inputs(data_dir + 'train_body_vecs.npy')
-decoder_input_data, decoder_target_data = load_decoder_inputs(data_dir + 'train_title_vecs.npy')
+  num_encoder_tokens, body_pp = load_text_processor(data_dir + 'body_pp.dpkl')
+  num_decoder_tokens, title_pp = load_text_processor(data_dir + 'title_pp.dpkl')
 
-num_encoder_tokens, body_pp = load_text_processor(data_dir + 'body_pp.dpkl')
-num_decoder_tokens, title_pp = load_text_processor(data_dir + 'title_pp.dpkl')
+  #arbitrarly set latent dimension for embedding and hidden units
+  latent_dim = 300
 
-#arbitrarly set latent dimension for embedding and hidden units
-latent_dim = 300
+  ##### Define Model Architecture ######
 
-##### Define Model Architecture ######
+  ########################
+  #### Encoder Model ####
+  encoder_inputs = tf.keras.layers.Input(shape=(doc_length,), name='Encoder-Input')
 
-########################
-#### Encoder Model ####
-encoder_inputs = tf.keras.layers.Input(shape=(doc_length,), name='Encoder-Input')
+  # Word embeding for encoder (ex: Issue Body)
+  x = tf.keras.layers.Embedding(
+        num_encoder_tokens, latent_dim, name='Body-Word-Embedding', mask_zero=False)(encoder_inputs)
+  x = tf.keras.layers.BatchNormalization(name='Encoder-Batchnorm-1')(x)
 
-# Word embeding for encoder (ex: Issue Body)
-x = tf.keras.layers.Embedding(
-      num_encoder_tokens, latent_dim, name='Body-Word-Embedding', mask_zero=False)(encoder_inputs)
-x = tf.keras.layers.BatchNormalization(name='Encoder-Batchnorm-1')(x)
+  # Intermediate GRU layer (optional)
+  #x = GRU(latent_dim, name='Encoder-Intermediate-GRU', return_sequences=True)(x)
+  #x = BatchNormalization(name='Encoder-Batchnorm-2')(x)
 
-# Intermediate GRU layer (optional)
-#x = GRU(latent_dim, name='Encoder-Intermediate-GRU', return_sequences=True)(x)
-#x = BatchNormalization(name='Encoder-Batchnorm-2')(x)
+  # We do not need the `encoder_output` just the hidden state.
+  _, state_h = tf.keras.layers.GRU(latent_dim, return_state=True, name='Encoder-Last-GRU')(x)
 
-# We do not need the `encoder_output` just the hidden state.
-_, state_h = tf.keras.layers.GRU(latent_dim, return_state=True, name='Encoder-Last-GRU')(x)
+  # Encapsulate the encoder as a separate entity so we can just
+  #  encode without decoding if we want to.
 
-# Encapsulate the encoder as a separate entity so we can just
-#  encode without decoding if we want to.
-encoder_model = tf.keras.Model(inputs=encoder_inputs, outputs=state_h, name='Encoder-Model')
+  # TODO(jlewi): I commented out the following two lines
+  # encoder_model = tf.keras.Model(inputs=encoder_inputs, outputs=state_h, name='Encoder-Model')
```

Comment:
@inc0 @hamelsmu It doesn't look like seq2seq_encouder_out is used anywhere; can we delete these two lines of code?

Reply:
@jlewi the seq2seq_encoder out __should__ be used to initialize the state of the decoder.  However, that code has been commented out, apparently as an interim step by someone?  See my comments below

Reply:
I commented this line and changed the line below.